server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/IdentitySetContract.scala

```diff
@@ -1963,7 +1963,7 @@ trait IdentitySetContract {
            |                        ],
            |                        "textSignature": "Difference text signature",
            |                        "htmlSignature": "<p>Difference html signature</p>",
-           |                        "mayDelete": true
+           |                        "mayDelete": false
```

Comment:
Are we sure in DB this value is `false`?

This fix with merge code seems to only apply to the `list` API. While in the DB likely the `mayDelete` was stored as `true`.

I feel that "incorrect". Likely the custom identity derived from server identity should be stored with `mayDelete=false` at the beginning rather than cheating at the retrieval phase. cc @vttranlina opinion on this?

BTW I see another issue: WHEN James deletes a server identity (e.g. an alias `bob-alias@domain.tld`), then the updated custom identity would not be able to be cleaned up, **BOB** would keep sending mail with the alias while RRT would reject.
Solution: Deletion hook interface upon server identities being deleted?

Reply:
> BTW I see another issue

Likely not happen with the TMail client as it only works with `mayDelete=true` identities. But likely we should solve the potential issue one day.

Reply:
The logic of "merging" will happen when the duplicate identity Id,
When that we will prefer to pick customIdentity. 
Because customIdentity and serverIdentity has same Id => `mayDelete=false` is hotFix correct

> Are we sure in DB this value is false?

maybe, the value in database for this case should be "false". (now it maybe `true` because bug)